packages/admin/admin-rte/src/core/translation/ToolbarButton.tsx

```diff
@@ -0,0 +1,45 @@
+import { useContentTranslationService } from "@comet/admin";
+import { Translate } from "@comet/admin-icons";
+import Tooltip from "@mui/material/Tooltip";
+import { convertToRaw } from "draft-js";
+import * as React from "react";
+import { FormattedMessage } from "react-intl";
+
+import ControlButton from "../Controls/ControlButton";
+import { IControlProps } from "../types";
+import { transformStateToXml } from "./xml/transformStateToXml";
+import { translateAndTransformXmlToState } from "./xml/translateAndTransformToState";
+
+function ToolbarButton({ editorState, setEditorState }: IControlProps): React.ReactElement {
+    const translationContext = useContentTranslationService();
+
+    async function handleClick(event: React.MouseEvent) {
+        if (translationContext) {
+            event.preventDefault();
+
+            const contentState = editorState.getCurrentContent();
+
+            const xml = transformStateToXml(contentState);
+
+            const translationPromises = xml.map(async (item) => ({
+                original: item,
+                replaceWith: (await translationContext.translate(item)) ?? item,
+            }));
```

Comment:
This shouldn't be done for each item, but for the complete XML

Reply:
@johnnyomair "xml" might be a bad name for the array.

![Screenshot 2024-01-26 at 11 00 41](https://github.com/vivid-planet/comet/assets/118435139/f4683d4e-2a02-493e-b393-5e7625e7b5c4)

![Screenshot 2024-01-26 at 11 00 27](https://github.com/vivid-planet/comet/assets/118435139/2ba3b8dd-f9b7-4de3-8965-488cc246d19b)

Or do you mean sending everything in the rte at the translation function at once?

![Screenshot 2024-01-26 at 10 55 39](https://github.com/vivid-planet/comet/assets/118435139/243dde7e-8b16-48f7-a10c-4cf1276d3217)

![Screenshot 2024-01-26 at 10 55 52](https://github.com/vivid-planet/comet/assets/118435139/8ed365f9-540e-40c4-87c3-534c7a6445b4)

![Screenshot 2024-01-26 at 10 48 17](https://github.com/vivid-planet/comet/assets/118435139/27b203a8-b236-40e3-b0ea-fe6941b6a727)

Reply:
> Or do you mean sending everything in the rte at the translation function at once?

This.

Reply:
Think you can't do this without string manipulation.

To sent it all at once i will need to join the xml array with an unique character / sequence that gets ignored by the translator. Then split the result and if the length of the split and if the length of the new array does not match the xml array, the translation is damaged ...

Any other recommendation for the splitting sequence instead of <split />?

Or is this fine f5bacdd7a4af927c592fa06c1f773aa1d2f4ba06?

![Screenshot 2024-01-30 at 07 44 49](https://github.com/vivid-planet/comet/assets/118435139/c7e84b93-f111-43d8-aa5e-77e4e4779b06)

Reply:
> Any other recommendation for the splitting sequence instead of ?

Instead of adding `<split/>` I'd wrap every block in a `<block>...</block>` tag.